tests/model/test_model.py

```diff
@@ -1221,6 +1221,51 @@ def test_update_star_status(self, mocker, model, event_op,
             assert (model.index['messages'][unchanged_id]['flags'] ==
                     flags_before)
 
+    @pytest.mark.parametrize('event_message_ids, indexed_ids', [
+        ([1], [1]),
+        ([1, 2], [1]),
+        ([1, 2], [1, 2]),
+        ([1], [1, 2]),
+        ([], [1, 2]),
+        ([1, 2], []),
+    ])
+    @pytest.mark.parametrize('event_op, flags_before, flags_after', [
+        ('add', [], ['read']),
+        ('add', ['read'], ['read']),
+        ('add', ['starred'], ['starred', 'read']),
+        ('add', ['read', 'starred'], ['read', 'starred']),
+    ])
+    def test_update_read_status(self, mocker, model, event_op,
+                                event_message_ids, indexed_ids,
+                                flags_before, flags_after):
+        model.index = dict(messages={msg_id: {'flags': flags_before}
+                                     for msg_id in indexed_ids})
+        event = {
+            'messages': event_message_ids,
+            'type': 'update_message_flags',
+            'flag': 'read',
+            'operation': 'add',
+            'all': False,
+        }
+
+        mocker.patch('zulipterminal.model.Model.update_rendered_view')
+        set_count = mocker.patch('zulipterminal.model.set_count')
+
+        model.update_message_flag_status(event)
+
+        changed_ids = set(indexed_ids) & set(event_message_ids)
+        for changed_id in changed_ids:
+            assert model.index['messages'][changed_id]['flags'] == flags_after
+        (model.update_rendered_view.
+         has_calls([mocker.call(changed_id) for changed_id in changed_ids]))
```

Comment:
This should be `assert_has_calls`, which is also a bug in the other test.

Reply:
Fixed